test/CMakeLists.txt

```diff
@@ -2,7 +2,7 @@ add_subdirectory(gtest)
 
 set(TEST_MAIN_SRC test-main.cc gtest-extra.cc gtest-extra.h util.cc)
 add_library(test-main STATIC ${TEST_MAIN_SRC})
-target_link_libraries(test-main gtest fmt)
+target_link_libraries(test-main PUBLIC gtest PRIVATE fmt)
```

Comment:
Otherwise the traditional library `fmt.lib` will be linked to the `module-test` executable **in addition** to the module library with the object artifact. CMake usage requirements of link libraries are transitive unless marked `PRIVATE`. It took me quite a while to figure out why `build.ninja` was pulling in both libraries in the link step of `module-test`. This is less of a problem in compilers which implement the "strong ownership" model (like msvc), but it would cause link failures in compilers following the "weak ownership" model (like gcc or clang) because of multiple definitions of the same linker symbols.